### PR TITLE
Cumulus-1962 Adding location params for rec reports

### PR DESCRIPTION
**Pull request by kkelly51**
**Summary:** Adding location params for rec reports

Addresses [CUMULUS-1962: Adding location params for rec reports](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1962)

## PR Checklist

- [x] Update CHANGELOG
- [x ] Unit tests
- [ ] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

**Review by flamingbear — CHANGES_REQUESTED** [review]
So I don't love the name 'location', but can't think of a better one. so 🤷 
Looks like what I was going to do. But we do need to have the location in the report output (and now that I say that probably add it to the test [which means my old tests should have their names added like for collectionId, but I can do that.])

File: packages/api/tests/lambdas/test-create-reconciliation-report.js [hunk 1]
```diff
@@ -748,6 +748,107 @@ test.serial(
...
+      GranuleFilesCache.batchUpdate({ puts: files }),
+    ]);
+
+    // Create collections that are in sync
+    const matchingColls = range(10).map(() => ({
+      name: randomId('name'),
+      version: randomId('vers'),
+    }));
+
+    await storeCollectionsToElasticsearch(matchingColls);
```
[the remark this answers is not in the document]
Comment (kkelly51) [reply packages/api/tests/lambdas/test-create-reconciliation-report.js:785]: ah no.

File: packages/api/lambdas/create-reconciliation-report.js [hunk 2]
```diff
@@ -605,14 +607,28 @@ async function createReconciliationReport(recReportParams) {
...
+    report = {
+      ...initialReportHeader(recReportParams),
+      collectionsInCumulusCmr: cloneDeep(reportFormatCumulusCmr),
+      granulesInCumulusCmr: cloneDeep(reportFormatCumulusCmr),
+      filesInCumulusCmr: cloneDeep(reportFormatCumulusCmr),
+    };
+  } else {
+    report = {
+      ...initialReportHeader(recReportParams),
+      filesInCumulus,
```
[the remark this answers is not in the document]
Comment (kkelly51) [reply packages/api/lambdas/create-reconciliation-report.js:626]: Yes, will bring it up.

File: packages/api/lambdas/create-reconciliation-report.js [hunk 3]
```diff
@@ -605,14 +607,28 @@ async function createReconciliationReport(recReportParams) {
...
-    collectionsInCumulusCmr: cloneDeep(reportFormatCumulusCmr),
-    granulesInCumulusCmr: cloneDeep(reportFormatCumulusCmr),
-    filesInCumulusCmr: cloneDeep(reportFormatCumulusCmr),
-  };
+  let report;
+  if (location === 'S3') {
+    report = {
+      ...initialReportHeader(recReportParams),
+      filesInCumulus,
+    };
```
[the remark this answers is not in the document]
Comment (kkelly51) [reply packages/api/lambdas/create-reconciliation-report.js]: added to header

**Review by flamingbear — APPROVED** [review]
LGTM 👍

**Pull request merged**